tests/foreman/api/test_remoteexecution.py

```diff
@@ -0,0 +1,107 @@
+"""Test for Remote Execution
+
+:Requirement: Upgraded Satellite
+
+:CaseAutomation: Automated
+
+:CaseLevel: Acceptance
+
+:CaseComponent: API
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+from nailgun import client
+from nailgun import entities
+from nailgun.entity_mixins import TaskFailedError
+
+from robottelo.api.utils import wait_for_tasks
+from robottelo.config import settings
+from robottelo.decorators import tier3
+from robottelo.decorators import tier4
+from robottelo.helpers import add_remote_execution_ssh_key
+from robottelo.vm_capsule import CapsuleVirtualMachine
+
+
+@tier4
+def test_positive_run_capsule_upgrade_playbook():
+    """Run Capsule Upgrade playbook against an External Capsule
+
+    :id: 9ec6903d-2bb7-46a5-8002-afc74f06d83b
+
+    :steps:
+        1. Create a Capsule VM, add REX key.
+        2. Run the Capsule Upgrade Playbook.
+
+    :expectedresults: Capsule is upgraded successfully
+
+    :CaseImportance: Medium
+    """
+    with CapsuleVirtualMachine() as capsule_vm:
+        add_remote_execution_ssh_key(capsule_vm.ip_addr)
+        job = entities.JobInvocation().run(
+            synchronous=False,
+            data={
+                'job_template_id': 170,
+                'inputs': {
+                    'target_version': "6.8.z",
+                    'whitelist_options': "repositories-validate,repositories-setup",
+                },
+                'targeting_type': "static_query",
+                'search_query': f"name = {capsule_vm.hostname}",
+            },
+        )
+        wait_for_tasks(f"resource_type = JobInvocation and resource_id = {job['id']}")
+        result = entities.JobInvocation(id=job['id']).read()
+        assert result.succeeded == 1
```

Comment:
Can we add more asserts, maybe by checking the actual capsule?

Reply:
We can add feature refresh or foreman-maintain health check to ensure the caps is running ok after the upgrade.